src/models/propulsion/FGTurboProp.h

```diff
@@ -176,7 +176,7 @@ class FGTurboProp : public FGEngine
 
   bool EngStarting;            // logicaly output - TRUE if engine is starting
   bool GeneratorPower;
-  int Condition;
+  bool Condition;
```

Comment:
This type modification is inconsistent with the code that uses `Condition`:
https://github.com/JSBSim-Team/jsbsim/blob/e61d86e02830d7380e1ba6ace1f3c470c4adf0a4/src/models/propulsion/FGTurboProp.cpp#L240-L243
https://github.com/JSBSim-Team/jsbsim/blob/e61d86e02830d7380e1ba6ace1f3c470c4adf0a4/src/models/propulsion/FGTurboProp.cpp#L255-L267

Reply:
Ok. Is that mean Condition must be set as integer in also UnrealEngine5 right?

Reply:
If you okey with Condition as integer I can apply this patch
```
diff --git a/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h b/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h
index d02bdfaf..1d6ac817 100644
--- a/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h
+++ b/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h
@@ -187,7 +187,7 @@ struct FEngineCommand
        UPROPERTY(BlueprintReadWrite, EditAnywhere)
        bool GeneratorPower = false;
        UPROPERTY(BlueprintReadWrite, EditAnywhere)
-       bool Condition = false;
+       int Condition = 0;


        FString GetDebugMessage()
diff --git a/src/models/propulsion/FGTurboProp.h b/src/models/propulsion/FGTurboProp.h
index c631f0b9..e599077a 100644
--- a/src/models/propulsion/FGTurboProp.h
+++ b/src/models/propulsion/FGTurboProp.h
@@ -124,14 +124,14 @@ public:
   double getOilTemp_degF (void) {return KelvinToFahrenheit(OilTemp_degK);}

   inline bool GetGeneratorPower(void) const { return GeneratorPower; }
-  inline bool GetCondition(void) const { return Condition; }
+  inline int GetCondition(void) const { return Condition; }

   void SetPhase( phaseType p ) { phase = p; }
   void SetReverse(bool reversed) { Reversed = reversed; }
   void SetCutoff(bool cutoff) { Cutoff = cutoff; }

   inline void SetGeneratorPower(bool gp) { GeneratorPower=gp; }
-  inline void SetCondition(bool c) { Condition=c; }
+  inline void SetCondition(int c) { Condition=c; }
   int InitRunning(void);
   std::string GetEngineLabels(const std::string& delimiter);
   std::string GetEngineValues(const std::string& delimiter);
@@ -176,7 +176,7 @@ private:

   bool EngStarting;            // logicaly output - TRUE if engine is starting
   bool GeneratorPower;
-  bool Condition;
+  int Condition;
   int thrusterType;            // the attached thruster

   double Off(void);
```
Finally it will shown like that.
![image](https://user-images.githubusercontent.com/22801690/216767414-ede89823-2809-4742-bbb8-581346c593cf.png)

Reply:
Could you please push that change to your branch ? That would help reviewing it.
Thanks.

Reply:
The issue regarding bool to int conversation, I do not think there is any bug. I think @Zaryob is trying to solve problems with the C130 aircraft in UE5. But the C130 has an turboprop engine last updated in 2011 with Aero-Matic v 0.8, engine is [t56.xml](https://github.com/JSBSim-Team/jsbsim/blob/e61d86e02830d7380e1ba6ace1f3c470c4adf0a4/engine/t56.xml). I found this t56 engine doesn't work in UE5 (and probably not JSBSim standalone either), so I tested newer turboprop engine [PW125BX](https://github.com/JSBSim-Team/jsbsim/blob/e61d86e02830d7380e1ba6ace1f3c470c4adf0a4/engine/PW125BX.xml) and this one works fine and c130 can fly. (although it is not correct thrust/power, i just test as example).

Reply:
Also looking closer, I see you were trying to fix/add code to plugin at line 1024 as seen your error `UnrealEngine\Plugins\JSBSimFlightDynamicsModel\Source\JSBSimFlightDynamicsModel\Private\JSBSimMovementComponent.cpp(1024): error C4800: Implicit conversion from 'int' to bool.`

If you want to fix the code at that location you should do something like this to avoid the error:
`      EngineStates[i].Condition = (TurboPropEngine->GetCondition() > 0);`
This allows for simple conversation from int to bool for UE5.
There's no need for changes to JSBSim or the plugin, besides this.

Reply:
Ok I will make revelant changes